server/src/main/java/org/apache/druid/server/coordinator/DruidCluster.java

```diff
@@ -92,6 +109,11 @@ public List<ServerHolder> getAllServers()
     return allServers;
   }
 
+  public List<ServerHolder> getAllManagedServers()
```

Comment:
I think we can completely remove `getAllServers()` now since the only usage is in `PrepareBalancerAndLoadQueues` to cancel loads if the server is decommissioning.

This usage should also be replaced with `getAllManagedServers()`.
So that a clone is just a clone and nothing else.
Even if someone accidentally marks a server both as a clone and as a decommissioning server, we will still treat it as a clone and not perform any of the decomm logic.

Reply:
Removed `allServers`